src/retrieveDefaultValues.js

```diff
@@ -0,0 +1,9 @@
+exports.retrieveDefaultValues = (schema, instance) => Object.keys(schema).reduce((values, prop) => {
+  const defaultDecl = schema[prop].default;
+
+  if (defaultDecl) {
+    values[prop] = typeof defaultDecl === 'function' ? defaultDecl(instance) : defaultDecl;
+  }
+
+  return values;
+}, {});
```

Comment:
Can we name this file `defaultValueAssignment` and then call the function `assignDefaultValue` to keep the pattern of naming the file/folder as a concept and the function an action?